command: add assets command to build statics and/or assets independently

ppanero opened the pull request:
Changes:
- Adds an `assets` command.
- Allows collecting statics
- Allows building with webpack
- Allows both of the previous actions at the same time.

Caveats:
- The `webpack buidlall` part displays the full logs on the terminal. It is due to some internal part of it printing to stdout and now allowing control. It will be most likely be fixed by https://github.com/pallets/click/issues/737 and/or when we manage to fix https://github.com/inveniosoftware/invenio-cli/issues/66 using an interpreter.
- Not implemented for `--prod`. Since it requires a restart of uwsgi. I assume the workflow for this would be to `update` statics and rebuild the app container (not base) if not. All after having tested/modified in a `--dev` environment. (No ?)

Closes https://github.com/inveniosoftware/invenio-cli/issues/24

fenekku commented:
> Not implemented for --prod [...] workflow for this would be to update statics and rebuild the app container (not base) if not

Yes. This should be documented somewhere so it is clear: the workflow is edit your local work and run --dev. When ready generate --prod.

fenekku commented on invenio_cli/helpers/scripts.py:
```diff
@@ -101,17 +101,32 @@ def bootstrap(log_config, dev=True, pre=True,
+    if webpack:
+        # Build using webpack
+        run_command(cli, runner, "webpack buildall",
+                    message="Building assets...", verbose=log_config.verbose)
```
Shelved: It would be nice to break up, dependency (npm) installation and asset building too.

fenekku commented on invenio_cli/helpers/scripts.py:
```diff
@@ -101,17 +101,32 @@ def bootstrap(log_config, dev=True, pre=True,
+def _build_container_assets(log_config, statics=True, webpack=True):
+    """Build assets in the containers."""
+    # TODO: Check how to address this. Requires uwsgi restart.
+    pass
```
Normal: Let's display a message that this is not something to do in prod / a full prod rebuild is what will take these changes into account. Important part here is to provide feedback to the user as opposed to no feedback.

ppanero replied:
True :) fixed. Example output:
![image](https://user-images.githubusercontent.com/6756943/70517952-c6c35f80-1b39-11ea-9b6b-81e9f4135620.png)

fenekku replied:
Nice!

ppanero commented:
Created issues / addressed comments.

> Yes. This should be documented somewhere so it is clear: the workflow is edit your local work and run --dev. When ready generate --prod.

Agree, will start writing something. Maybe we can start creating more pages in the Wiki until we have a doc tool.
